libs/GraphicECS/SFML/Resources/MusicResource.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+** EPITECH PROJECT, 2022
+** R-Type
+** File description:
+** MusicResource
+*/
+
+#ifndef MUSICRESOURCE_HPP_
+#define MUSICRESOURCE_HPP_
+
+#include <SFML/Audio.hpp>
+#include <filesystem>
+#include "Resource/Resource.hpp"
+#include <unordered_map>
+#include "Error/Error.hpp"
+
+namespace graphicECS::SFML::Resources
+{
+    /// @brief This resource class stores a map of Musics SFML.
+    class MusicResource : public ecs::Resource {
+      public:
+        /// @brief Enumeration of all available Musics
+        enum musicName_e { UNDEFINED, MENUTHEME, COMBATTHEME };
+
+        /// @brief Name of map which contains Musics.
+        using MusicsList = std::unordered_map<musicName_e, std::shared_ptr<sf::Music>>;
+
+        /// @brief Add a Music from it's Music Path passed as parameter
+        /// @param music_e Enum of the Music
+        /// @param musicPath The music path to be used.
+        inline MusicResource(const musicName_e music_e, const std::filesystem::path &musicPath)
```

Comment:
May add a possibility to create a resource without music and add them later (example : if you have levels : the level 1 doesn't use the music of the other level so they don't have to be loaded) ?

Reply:
Nop all song and sounds must be loaded at the creation of the world in the music/sound Resource

Reply:
Why must they be created at initialization?

Reply:
Music are loaded at start like texture, fonts...